apps/fz_http/lib/fz_http_web/auth/json/authentication.ex

```diff
@@ -0,0 +1,28 @@
+defmodule FzHttpWeb.Auth.JSON.Authentication do
+  @moduledoc """
+  API Authentication implementation module for Guardian.
+  """
+  use Guardian, otp_app: :fz_http
```

Comment:
Just a general comment not related to this PR: I honestly don't feel secure using a library for authentication, it can change any day and some developers can bump the version without looking into the diff. I guess that's why Phoenix provides a generator for it instead of having a library.

Reply:
Guardian helped us close the seed by integrating Google and Okta in record time, but I'm certainly in favor of rolling our own if we're confident we can Do It Right™.

We could fork it to our org to keep it pinned to a version we control, but this risks going stale (another source of security problems) and diverging too far from upstream.

For something as simple as JWT encode and decode, where `Phoenix.Token` abstracts things nicely for us, seems like a good opportunity to replace Guardian. For OAuth2 / OIDC I think shouldn't be too bad either. SAML though... good luck 😅

Reply:
Opened https://github.com/firezone/product/issues/548 to track further discussion.

Re: the version bump problem -- Maybe having some convention around version bumps would be a good idea. Dependabot links the diff in the PRs it opens, but maybe there's a way to configure it to *require* admin reviews for PRs that bump certain defined dependencies.

Supply chain has seen a lot of new Best Practices™ developed after the SolarWinds attack. I have some experience in this area -- can followup over chat.

Reply:
We can leverage https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners for `mix.exs` and `mix.lock`, and require code owners review to merge PR.